errorsbp/prefix.go

```diff
@@ -0,0 +1,48 @@
+package errorsbp
+
+// PrefixError appends prefix to err.
+//
+// If err is nil, nil will be returned.
+// If prefix is empty, err will be returned as-is.
+// Otherwise, the returned error will be of type *PrefixedError,
+// and the error message would be:
+//
+//     "prefix: err.Error()"
+//
+// NOTE: A reason to use PrefixError over fmt.Errorf(prefix + ": %w", err)
+// is that prefix could contain format verbs, e.g. prefix = "foo%sbar".
+func PrefixError(prefix string, err error) error {
```

Comment:
What is the benefit of `PrefixError` over using `fmt.Errorf` with `%w`? I am generally a fan of using the stdlib features over making our own.

Reply:
with `fmt.Errorf` you can't blindly do `errs <- fmt.Errorf("prefix: %w", worker())`, because that will convert a nil-error into a non-nil error.

Reply:
another benefit is that if you want to check which worker failed, you can (by `errors.As(err, new(errorbp.PrefixedError))` then check its `Prefix()`).

Reply:
If you're looking to enable figuring out which worker failed, make this a WorkerError.  Don't conflate a prefix with the worker that failed.